ast-nodes/expressions/operators/div-operator.dart

```diff
@@ -29,6 +33,24 @@ class DivOperator extends BinaryRelation implements Product {
   }
 
   void checkSemantics() {
-    // TODO: implement
+    leftOperand.checkSemantics();
+    rightOperand.checkSemantics();
+
+    if ((leftOperand.resultType is RealType &&
+            rightOperand.resultType is IntegerType) ||
```

Comment:
Well, exactly. I think it must be illegal to divide `1.0` by `true`, but if we leave `leftOperand.resultType is RealType || rightOperand.resultType is RealType` only, it will cast `true` to `1.0`. Dividing real by bool and vice versa doesn't make much sense.

Reply:
It makes sense semantically and it might make sense for the programmer, our job is mostly not to create unexpected restrictions. Let's allow `real / boolean`

Reply:
Should we allow `bool/bool`? And what is the result type of such operation?

Reply:
Integer, but both operands should be casted to integer first